chia/wallet/wallet_node.py

```diff
@@ -225,6 +227,60 @@ async def get_private_key(self, fingerprint: Optional[int]) -> Optional[PrivateK
 
         return key
 
+    def set_resync_flag(self, fingerprint, flag: bool = True):
+        config = load_config(self.root_path, "config.yaml")
+        if flag is True:
+            config["wallet"]["reset_sync_for_fingerprint"] = fingerprint
+            self.log.info("Enabled resync for wallet fingerprint: %s", fingerprint)
+        else:
+            self.log.debug(
+                "Trying to disable resync: %s [%s]", fingerprint, config["wallet"].get("reset_sync_for_fingerprint")
+            )
+            if config["wallet"].get("reset_sync_for_fingerprint") == fingerprint:
+                del config["wallet"]["reset_sync_for_fingerprint"]
+                self.log.info("Disabled resync for wallet fingerprint: %s", fingerprint)
+        save_config(self.root_path, "config.yaml", config)
+
+    async def reset_sync_db(self, db_path, fingerprint):
+        conn: aiosqlite.Connection
+        async with manage_connection(db_path) as conn:
+            self.log.info("Resetting wallet sync data...")
+            rows = list(await conn.execute_fetchall("SELECT name FROM sqlite_master WHERE type='table'"))
+            try:
+                # if less than 20 it means we're running on old database and that's
+                # handled, but not if there's more
+                assert len(rows) < 20, (
+                    f"Expected 20 tables, found {len(rows)} tables."
+                    " Please check if the new table needs to be added in clean_resync_tables."
+                )
+            except AssertionError:
+                self.log.exception("Incompatible database to reset")
+                return
+            await conn.execute("BEGIN")
+            commit = True
+            tables = [row[0] for row in rows]
+            try:
+                if "coin_record" in tables:
+                    await conn.execute("DELETE FROM coin_record")
+                if "interested_coins" in tables:
+                    await conn.execute("DELETE FROM interested_coins")
+                if "interested_puzzle_hashes" in tables:
+                    await conn.execute("DELETE FROM interested_puzzle_hashes")
+                if "key_val_store" in tables:
+                    await conn.execute("DELETE FROM key_val_store")
+                if "users_nfts" in tables:
+                    await conn.execute("DELETE FROM users_nfts")
+            except aiosqlite.Error:
+                self.log.exception("Error resetting sync tables")
+                await conn.execute("ROLLBACK")
```

Comment:
nit: I would probably move the rollback into the finally block in an else statement if commit is false.